src/contract.rs

```diff
@@ -914,13 +999,120 @@ fn query_validators_metrics_by_timestamp(
     Ok(res)
 }
 
+
+// off chain code
+
+//todo: test
+fn remove_off_chain_metrics_for_timestamp(deps: DepsMut, info: MessageInfo, timestamp: u64)
+                                          -> Result<Response, ContractError> {
+
+    if !sender_is_manager(&deps, &info) {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    let mut off_chain_timestamps = OFF_CHAIN_TIMESTAMPS.load(deps.storage)?;
+
+    let existing_timestamps_length = off_chain_timestamps.timestamps.len();
+
+    let new_timestamps = off_chain_timestamps.timestamps.into_iter()
```

Comment:
Simplify all of this logic by making timestamp a mapping between index <-> timestamp

Reply:
Overall the usage for remove_timestamp will be very limited.